### PR TITLE
Add auth header support for tracer exporter

### PR DESCRIPTION
**Pull request by srijan-27**
- Add support for auth header in `zipkin` and `jaeger` exporters.
- Authorization header can be passed in `TRACER_AUTH_KEY` config in .env file.
- Closes: #729

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

### CODE REVIEW

File: pkg/gofr/gofr.go [hunk 1]
```diff
@@ -343,6 +321,46 @@ func (a *App) initTracer() {
...
+				otlptracegrpc.WithHeaders(map[string]string{"Authorization": authHeader}))
+		} else {
+			exporter, err = otlptracegrpc.New(context.Background(), otlptracegrpc.WithInsecure(),
+				otlptracegrpc.WithEndpoint(fmt.Sprintf("%s:%s", host, port)))
+		}
+	case "zipkin":
+		a.container.Log("Exporting traces to zipkin.")
+
+		if authHeader != "" {
+			exporter, err = zipkin.New(fmt.Sprintf("http://%s:%s/api/v2/spans", host, port),
```
[the remark this answers is not in the document]
Comment (srijan-27) [reply pkg/gofr/gofr.go:348]: are you referring to line 348?

File: pkg/gofr/gofr.go [hunk 2]
```diff
@@ -343,6 +321,46 @@ func (a *App) initTracer() {
...
+				zipkin.WithHeaders(map[string]string{"Authorization": authHeader}))
```
[the remark this answers is not in the document]
Comment (srijan-27) [reply pkg/gofr/gofr.go:349]: you can refer: https://pkg.go.dev/go.opentelemetry.io/otel/exporters/zipkin@v1.28.0#New

File: pkg/gofr/gofr.go [hunk 3]
```diff
@@ -343,6 +321,46 @@ func (a *App) initTracer() {
...
+		} else {
+			exporter, err = zipkin.New(fmt.Sprintf("http://%s:%s/api/v2/spans", host, port))
+		}
+	case traceExporterGoFr:
+		exporter = NewExporter("https://tracer-api.gofr.dev/api/spans", logging.NewLogger(logging.INFO))
+
+		a.container.Log("Exporting traces to GoFr at https://tracer.gofr.dev")
```
[the remark this answers is not in the document]
Comment (srijan-27) [reply pkg/gofr/gofr.go:356]: yes, it is known and will be fixed in a separate PR alongwith custom endpoints support for traces.

File: pkg/gofr/gofr.go [hunk 4]
```diff
@@ -343,6 +321,46 @@ func (a *App) initTracer() {
...
+		err      error
+	)
+
+	authHeader := strings.TrimSpace(a.Config.Get("TRACER_AUTH_KEY"))
+
+	switch strings.ToLower(name) {
+	case "jaeger":
+		a.container.Log("Exporting traces to jaeger.")
+
+		if authHeader != "" {
```
[the remark this answers is not in the document]
Comment (srijan-27) [reply pkg/gofr/gofr.go:336]: If you look closely, you will observe that even if there is a line break, the statement is not ending there.. it continues even after it.

File: pkg/gofr/gofr.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccoVeille) [reply pkg/gofr/gofr.go:349]: I totally missed you were passing options, same as 

https://github.com/gofr-dev/gofr/pull/828#discussion_r1669821352

What about this then ?

```diff
-		if authHeader != "" {
-			exporter, err = zipkin.New(fmt.Sprintf("http://%s:%s/api/v2/spans", host, port),
-				zipkin.WithHeaders(map[string]string{"Authorization": authHeader}))
-		} else {
-			exporter, err = zipkin.New(fmt.Sprintf("http://%s:%s/api/v2/spans", host, port))
-		}
+             var opts []zipkin.Option
+		if authHeader != "" {
+			opts = append(opts, zipkin.WithHeaders(map[string]string{"Authorization": authHeader}))
+		}
+		exporter, err = zipkin.New(fmt.Sprintf("http://%s:%s/api/v2/spans", host, port), opts...)
```

File: pkg/gofr/gofr.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ccoVeille) [reply pkg/gofr/gofr.go:336]: I totally missed indeed.

What about this, then
```diff
-		if authHeader != "" {
-			exporter, err = otlptracegrpc.New(context.Background(), otlptracegrpc.WithInsecure(),
-				otlptracegrpc.WithEndpoint(fmt.Sprintf("%s:%s", host, port)),
-				otlptracegrpc.WithHeaders(map[string]string{"Authorization": authHeader}))
-		} else {
-			exporter, err = otlptracegrpc.New(context.Background(), otlptracegrpc.WithInsecure(),
-				otlptracegrpc.WithEndpoint(fmt.Sprintf("%s:%s", host, port)))
-		}
+		opts := []otlptracegrpc.Option{
+				otlptracegrpc.WithInsecure(),
+				otlptracegrpc.WithEndpoint(fmt.Sprintf("%s:%s", host, port)))
+		if authHeader != "" {
+				opts = append(opts, otlptracegrpc.WithHeaders(map[string]string{"Authorization": authHeader}))
+		}
+		exporter, err = otlptracegrpc.New(context.Background(), opts...)
```

**Review by ccoVeille — APPROVED** [review]

**Review by Umang01-hash — APPROVED** [review]

**Pull request merged**